packages/app-builder/src/routes/_builder+/decisions+/$decisionId.tsx

```diff
@@ -38,15 +42,22 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
   const { apiClient } = await authService.isAuthenticated(request, {
     failureRedirect: getRoute('/sign-in'),
   });
+  const parsedParam = await parseParamsSafe(
+    params,
+    z.object({ decisionId: shortUUIDSchema }),
+  );
+  if (!parsedParam.success) {
+    throw handleParseParamError(request, parsedParam.error);
+  }
+  const { decisionId } = parsedParam.data;
 
-  const decisionId = fromParams(params, 'decisionId');
   try {
     const decision = await apiClient.getDecision(decisionId);
 
     return json({ decision });
   } catch (error) {
     if (isNotFoundHttpError(error)) {
-      throw new Response(null, { status: 404, statusText: 'Not Found' });
+      throw json(null, NOT_FOUND);
```

Comment:
```suggestion
      return json(null, NOT_FOUND);
```
unless I misunderstand?

Reply:
got it 👍  ça me va